storage/storage.go

```diff
@@ -108,31 +109,55 @@ func (m multiStorage) Stat(ctx context.Context, element ingest.Identifiable) (*O
 	return o0, err.Err()
 }
 
-func (m multiStorage) Store(ctx context.Context, element ingest.Identifiable, download func(context.Context, ingest.Identifiable) (ingest.Object, error)) (*url.URL, error) {
+func (m multiStorage) Configure(map[string]any) error {
+	return errors.New("not implemented")
+}
+
+func (m multiStorage) Store(ctx context.Context, element ingest.Codec, obj ingest.Object) (*url.URL, error) {
 	var u0 *url.URL
 	ch := make(chan error, len(m))
+	// TODO: the whole copying could be improved, too many copies of the same data.
+	buf, err := io.ReadAll(obj.Reader)
```

Comment:
yes, this is really unfortunate. It might be possible to make this nicer if we transform the Store operation to return a `Writer` that the client writes to. This way the client could get some bytes once and write them to multiple destinations concurrently and then get more bytes. this could be accomplished by using a TeeReader (https://pkg.go.dev/io#TeeReader) and layering n-1 TeeReaders and then doing one `copy` from the resulting TeeReader to the final Writer.